### PR TITLE
Artp 375/notifications in browser

### PR DESCRIPTION
**Pull request by pdeleastar**
Notification pop up added for browser
- Notification appears when trade excecuted
- Trade info (notional, ccy, rate and trade ID) appears

### CODE REVIEW

File: src/client/src/rt-components/platform/adapters/browser/browser.ts [hunk 1]
```diff
@@ -11,4 +12,13 @@ export default class Browser implements PlatformAdapter {
 
     open: (config: WindowConfig, onClose: () => void) => openBrowserWindow(config, onClose),
   }
+
+  notification = {
+    notify: (message: any) => {
+      const { tradeNotification } = message
```
Comment (danmcdaid) [inline src/client/src/rt-components/platform/adapters/browser/browser.ts]: Would be better to define a type for message that types `tradeNotifications`

File: src/client/src/rt-components/platform/adapters/browser/browser.ts [hunk 2]
```diff
@@ -11,4 +12,13 @@ export default class Browser implements PlatformAdapter {
...
+      if (Notification.permission === 'granted') {
+        sendNotification(tradeNotification)
```
Comment (danmcdaid) [inline src/client/src/rt-components/platform/adapters/browser/browser.ts]: Are permissions enums? Or can they be?

File: src/client/src/ui/blotter/epics/blotterServiceEpic.ts [hunk 3]
```diff
@@ -60,4 +60,18 @@ const connectBlotterToNotifications: ApplicationEpic = (action$, state$, { platf
...
 
-export const publishBlotterEpic = combineEpics(connectBlotterToExcel, connectBlotterToNotifications)
+export const requestBrowserNotificationPermission: ApplicationEpic = (action$, state$, { platform }) =>
+  action$.pipe(
+    applicationConnected,
+    delay(3000),
+    tap(() => {
+      Notification.requestPermission(permission => {
+        if (permission === 'granted') {
+          console.log('Notification permission granted')
```
Comment (danmcdaid) [inline src/client/src/ui/blotter/epics/blotterServiceEpic.ts]: As above enum for permission value.  Also is this the required behaviour? Do we need to react to a permission not granted? Either way I'm questioning the console.log!

File: src/client/src/ui/blotter/epics/blotterServiceEpic.ts [hunk 4]
```diff
@@ -60,4 +60,18 @@ const connectBlotterToNotifications: ApplicationEpic = (action$, state$, { platf
     ignoreElements(),
   )
 
-export const publishBlotterEpic = combineEpics(connectBlotterToExcel, connectBlotterToNotifications)
+export const requestBrowserNotificationPermission: ApplicationEpic = (action$, state$, { platform }) =>
+  action$.pipe(
+    applicationConnected,
+    delay(3000),
```
Comment (danmcdaid) [inline src/client/src/ui/blotter/epics/blotterServiceEpic.ts]: Why a 3 second delay? This is waiting for some other async event to have taken place?

File: src/client/src/rt-components/platform/adapters/browser/browser.ts (hunk 1 above)
Comment (pdeleastar) [reply src/client/src/rt-components/platform/adapters/browser/browser.ts]: Added type for trade notification

File: src/client/src/rt-components/platform/adapters/browser/browser.ts (hunk 2 above)
Comment (pdeleastar) [reply src/client/src/rt-components/platform/adapters/browser/browser.ts]: Made an enum for permissions

File: src/client/src/ui/blotter/epics/blotterServiceEpic.ts (hunk 3 above)
Comment (pdeleastar) [reply src/client/src/ui/blotter/epics/blotterServiceEpic.ts]: This has been removed as it is not needed, it is just Notification.requestPermission() now

File: src/client/src/ui/blotter/epics/blotterServiceEpic.ts (hunk 4 above)
Comment (pdeleastar) [reply src/client/src/ui/blotter/epics/blotterServiceEpic.ts]: 3 second delay removed, it was not waiting for any async event, the notification permissions request now happens instantly

**Pull request merged**